javascript/Post/create.mjs

```diff
@@ -0,0 +1,41 @@
+import { fetchTokens } from "../Post/fetchTokens.mjs";
+import { API_BASE_URL } from "../constant/index.mjs";
+
+const action = "/api/v1/social/posts";
+
+
+/**
+ * This will create a post using new data from a form
+ * @param {object} createPostData This is the new data from a form
+ */
+export async function createPost(createPostData) {
+  const createPostsURL = API_BASE_URL + action;
+
+  const response = await fetchTokens(createPostsURL, {
+    method: "POST",
+    body: JSON.stringify(createPostData),
+  });
+  const post = await response.json();
+
+  if (post) {
+    return post;
+  } else {
+    console.warn("you are not allowed to edit this post");
+  }
+  return post;
+}
+
+function wait(time) {
+  return new Promise(function (res) {
+    setTimeout(res, time);
+  });
+}
+
+export async function waitBeforeExecute(post) {
+  createPost(post);
+
+  await wait(500);
```

Comment:
```suggestion
```

This is not reliable, you don't know how long it will take for this to finish. On your computer, it might be quick, but I am marking your work from Africa. I am further away from the source of the data, so it takes longer to arrive with me. 500 is not enough and causes an error, please use await instead.